### PR TITLE
ci: optimize image build

### PR DESCRIPTION
**Pull request by daixiang0**
Signed-off-by: Long Dai <long0dai@foxmail.com>

**What type of PR is this?**

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

File: hack/docker.sh [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+# This script holds docker related functions.
+
+REPO_ROOT=$(dirname "${BASH_SOURCE[0]}")/..
+REGISTRY=${REGISTRY:-"swr.ap-southeast-1.myhuaweicloud.com/karmada"}
+VERSION=${VERSION:="unknown"}
```
Comment (kevin-wangzefeng) [inline hack/docker.sh:11]: Can you (maybe in a separate PR) make default version to: latest tag + git_hash?

ref: https://github.com/kubernetes/kubernetes/blob/master/hack/lib/version.sh

Comment (daixiang0) [reply hack/docker.sh:11]: Sure.

**Review by kevin-wangzefeng — APPROVED** [review]
LGTM

File: hack/docker.sh [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+function build_images() {
+  local target="$1"
+  docker build -t ${REGISTRY}/${target}:${VERSION} -f ${REPO_ROOT}/cluster/images/${target}/Dockerfile ${REPO_ROOT}
```
Comment (RainbowMango) [inline hack/docker.sh:15]: Yeah, we don't have to copy binary to the docker file directory.

**Review by RainbowMango — APPROVED** [review]

**Pull request merged**